pandas/core/groupby/generic.py

```diff
@@ -1031,11 +1031,36 @@ def _cython_agg_blocks(
         agg_blocks: List[Block] = []
         new_items: List[np.ndarray] = []
         deleted_items: List[np.ndarray] = []
-        # Some object-dtype blocks might be split into List[Block[T], Block[U]]
-        split_items: List[np.ndarray] = []
-        split_frames: List[DataFrame] = []
 
         no_result = object()
+
+        def cast_result_block(result, block: "Block", how: str) -> "Block":
```

Comment:
can you put this function elsewhere (module level somewhere) and import, in-line functions make much harder to grok.

Reply:
this likely actually belongs as a method on block itself no?

Reply:
i dont think this merits a Block method.  There will be further refactoring coming up with the goal of having this iteration-over-blocks go through a BlockManager method

Reply:
@jreback is this a deal-breaker?  id like to postpone this decision until after implementing the BlockManager method that this/apply/ewm eventually call

Reply:
ok sounds good